test/e2e/aws_test.go

```diff
@@ -48,49 +59,75 @@ func (a *awsTestHelper) makeCredentialsSecret(namespace string) *corev1.Secret {
 	}
 }
 
+func (a *awsTestHelper) defaultExternalDNS(credsSecret *corev1.Secret) operatorv1alpha1.ExternalDNS {
+	return operatorv1alpha1.ExternalDNS{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testExtDNSName,
+		},
+		Spec: operatorv1alpha1.ExternalDNSSpec{
+			Zones: []string{hostedZoneID},
+			Source: operatorv1alpha1.ExternalDNSSource{
+				ExternalDNSSourceUnion: operatorv1alpha1.ExternalDNSSourceUnion{
+					Type: operatorv1alpha1.SourceTypeService,
+					Service: &operatorv1alpha1.ExternalDNSServiceSourceOptions{
+						ServiceType: []corev1.ServiceType{
+							corev1.ServiceTypeLoadBalancer,
+							corev1.ServiceTypeClusterIP,
+						},
+					},
+					AnnotationFilter: map[string]string{
+						"external-dns.mydomain.org/publish": "yes",
+					},
+				},
+				HostnameAnnotationPolicy: "Ignore",
+				FQDNTemplate:             []string{fmt.Sprintf("{{.Name}}.%s", hostedZoneDomain)},
+			},
+			Provider: operatorv1alpha1.ExternalDNSProvider{
+				Type: operatorv1alpha1.ProviderTypeAWS,
+				AWS: &operatorv1alpha1.ExternalDNSAWSProviderOptions{
+					Credentials: operatorv1alpha1.SecretReference{
+						Name: credsSecret.Name,
+					},
+				},
+			},
+		},
+	}
+}
 func (a *awsTestHelper) platform() string {
 	return string(configv1.AWSPlatformType)
 }
 
-func (a *awsTestHelper) ensureHostedZone(rootDomain string) (string, []string, error) {
+func (a *awsTestHelper) ensureHostedZone() (string, []string, error) {
 	zones, err := a.r53Client.ListHostedZones(&route53.ListHostedZonesInput{})
 	if err != nil {
 		return "", nil, fmt.Errorf("failed to list hosted zones: %w", err)
 	}
 
-	var (
-		nameServers  []string
-		hostedZoneID string
-	)
 	// if hosted zone exists then return its id and nameservers
 	for _, zone := range zones.HostedZones {
-		if aws.StringValue(zone.Name) == rootDomain {
+		if aws.StringValue(zone.Name) == hostedZoneDomain {
```

Comment:
See the previous comment about the dependency on the global variables.

Reply:
Done